src/Nest/CommonAbstractions/Infer/Field/FieldExpressionVisitor.cs

```diff
@@ -35,7 +41,7 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
 				var lastArg = node.Arguments.Last();
 				var constantExpression = lastArg as ConstantExpression;
 				this.Found = constantExpression == null;
-				return node;
```

Comment:
I don't think so.  We need to actually visit all of the inner nodes to ensure there are only constants in the expression.